src/core_modules/capture-core/components/Pages/EnrollmentEvent/EnrollmentEventPage.container.js

```diff
@@ -18,12 +23,33 @@ export const EnrollmentEventPage = () => {
         }),
         shallowEqual,
     );
-    const [mode] = useState(pageMode.VIEW);
     const { program } = useProgramInfo(programId);
     const programStage = [...program.stages?.values()].find(
         item => item.id === stageId,
     );
+
     return (
-        <EnrollmentEventPageComponent mode={mode} programStage={programStage} />
+        <EnrollmentEventPageComponent
+            mode={showEditEvent ? pageMode.EDIT : pageMode.VIEW}
+            programStage={programStage}
+            onEdit={() => onOpenEditEvent()}
```

Comment:
I think we should approach this more like you did at first. We should let the consumer of the `WidgetEventEdit` set the mode and then we deal with the complexity internally in the Widget. For example, the action `startShowEditEventDataEntry` we can call from an `useEffect` if the mode is set to edit. We still need the mode and onEdit callback here, but these will be values from useState.

Reply:
Hey @JoakimSM,

 The switch between the form edit/view mode comes from the redux store inside `state.viewEventPage.eventDetailsSection.showEditEvent`. This value is currently modified inside:

1. `EditEventDataEntry` with the `Save` button and the `Cancel` button
2. `WidgetEventEdit` with the Edit button (`dispatch` of `startShowEditEventDataEntry`)

If I understand you correctly, you suggest that the page `EnrollmentEventPage` should handle the form edit/view mode and store it inside its state. In this case, I would need to change the logic for the `Save` button and the `Cancel` button inside the `EditEventDataEntry`. I will probably need to remove their current logic that modifies `state.viewEventPage.eventDetailsSection.showEditEvent` and add it inside `EnrollmentEventPage` using callbacks or dispatches. The old logic will also have to be maintained as long as we still use this form for single events. 

I prefer to have `EditEventDataEntry`, `ViewEventDataEntry` and their parent `WidgetEventEdit` handle the form edit/view mode without involving the page. Let's have a chat via Slack when you are back to check what is the best strategy. 
Thank you!